compiler/rustc_resolve/src/lib.rs

```diff
@@ -2584,10 +2584,32 @@ impl<'a> Resolver<'a> {
                                 res_err = Some(CannotCaptureDynamicEnvironmentInFnItem);
                             }
                         }
-                        ConstantItemRibKind(_) => {
+                        ConstantItemRibKind(_, ident) => {
                             // Still doesn't deal with upvars
                             if record_used {
-                                self.report_error(span, AttemptToUseNonConstantValueInConstant);
+                                if let Some(ident) = ident {
+                                    let sugg = "consider using `let` instead of `const`";
+                                    self.report_error(
+                                        span,
+                                        AttemptToUseNonConstantValueInConstant(Some(ident), sugg),
+                                    );
+                                } else {
+                                    let resolutions = self.resolutions(self.graph_root).borrow();
```

Comment:
This is very incorrect.
At the call point of `validate_res_from_ribs` the necessary span is the key span of `ribs[i].bindings.get_key_value(&rib_ident)`, it can be passed from there to `validate_res_from_ribs` if necessary.
(Or it can be recalculated inside `validate_res_from_ribs` again, all the input data is also passed.)